parser/parser_test.go

```diff
@@ -639,23 +639,276 @@ func TestParseValidInput(t *testing.T) {
 			t.Errorf("test case: %s", testCase.input)
 			t.Errorf("length mismatch, expected %d definitions, got %d",
 				len(testCase.defs), len(file.Defs))
+
+type namedItem struct {
+	name string
+	item reflect.Value
+}
+
+// is supposed to tell whether two reflect.Value objects refer to objects that are equal
+func compareValues(a reflect.Value, b reflect.Value) (equal bool) {
+	equal, _ = deepCompareObjects(namedItem{"a", a}, namedItem{"b", b}, 0)
+	return equal
+}
+
+func findMatching(item reflect.Value, list []reflect.Value) (match reflect.Value, found bool) {
+	for _, x := range list {
+		if compareValues(x, item) {
+			return x, true
 		}
 
-		if len(file.Comments) == len(testCase.comments) {
-			for i := range file.Comments {
-				if !reflect.DeepEqual(file.Comments[i], testCase.comments[i]) {
-					t.Errorf("test case: %s", testCase.input)
-					t.Errorf("incorrect comment %d:", i)
-					t.Errorf("  expected: %s", testCase.comments[i])
-					t.Errorf("       got: %s", file.Comments[i])
-				}
+	}
+	return item, false
+}
+
+func isIn(item reflect.Value, list []reflect.Value) (contained bool) {
```

Comment:
I think this function (and findMatching and compareValues) are unnecessary due to comments below